File: cli/src/main/java/io/microconfig/cli/ssl/TrustManagerFactory.java
package io.microconfig.cli.ssl;

import io.microconfig.cli.CliException;

import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import java.io.File;
import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;

public class TrustManagerFactory {

    public static SSLContext trustAll() {
        try {
            var sc = SSLContext.getInstance("TLS");
            sc.init(null, new TrustManager[]{new AllowAllTrustManager()}, new java.security.SecureRandom());
            return sc;
        } catch (NoSuchAlgorithmException | KeyManagementException e) {
            throw new CliException("Can't init TLS context: " + e.toString(), 1000);
        }

    }

    public static SSLContext defaultTrust() {
        try {
            return SSLContext.getDefault();
        } catch (NoSuchAlgorithmException e) {
            throw new CliException("Can't init TLS context: " + e.toString(), 1000);
        }
    }

    public static SSLContext rootCa(String path) {
        try {
            var rootCa = new File(path);
            var sc = SSLContext.getInstance("TLS");
            var trustManager = new ProvidedCaTrustManager(rootCa);
            sc.init(null, new TrustManager[]{trustManager}, new java.security.SecureRandom());
            return sc;
        } catch (NoSuchAlgorithmException | KeyManagementException e) {
            throw new CliException("Can't init TLS context: " + e.toString(), 1000);
        }
    }

}

File: common/src/main/kotlin/io/microconfig/server/common/ssl/TrustManagerFactory.kt
package io.microconfig.server.common.ssl

import java.io.File
import java.security.KeyManagementException
import java.security.NoSuchAlgorithmException
import java.security.SecureRandom
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManager

fun trustAll(): SSLContext {
    return try {
        SSLContext.getInstance("TLS")
            .apply { init(null, arrayOf<TrustManager>(AllowAllTrustManager()), SecureRandom()) }
    } catch (e: NoSuchAlgorithmException) {
        throw RuntimeException("Can't init TLS context: $e")
    } catch (e: KeyManagementException) {
        throw RuntimeException("Can't init TLS context: $e")
    }
}

fun defaultTrust(): SSLContext {
    return SSLContext.getDefault()
}

fun rootCa(path: String): SSLContext {
    val rootCa = File(path)
    val sc = SSLContext.getInstance("TLS")
    val trustManager = ProvidedCaTrustManager(rootCa)
    sc.init(null, arrayOf<TrustManager>(trustManager), SecureRandom())
    return sc
}
